File: enrollment-service/src/main/java/com/learning/enrollment_service/entity/Course.java
package com.learning.enrollment_service.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Course {

    private Long id;

    private String title;

    private String description;

    private User instructor;

    private BigDecimal price = BigDecimal.ZERO;

    private String category;

    private String thumbnailUrl;

    private LocalDateTime createdAt = LocalDateTime.now();

}

File: my-app/src/types/Course.ts
export interface Instructor {
  id: number;
  username: string;
  fullName: string;
}

export interface Course {
  id: number;
  title: string;
  description?: string;
  price: number;
  thumbnailUrl?: string;
  category?: string;
  instructor: Instructor;
  createdAt: string;
}

export interface CoursePage {
  content: Course[];
  totalElements: number;
  totalPages: number;
}

export interface CourseCardProps extends Course {
  hours: number;  
  lessons: number;  
  level: string;  
  rating?: number;
}